api/app/models/resource.rb

```diff
@@ -63,6 +63,22 @@ class Resource < ApplicationRecord
     return all unless project.present?
     where(project: project)
   }
+  scope :by_tag, lambda { |tag|
+    return all unless tag.present?
+    tagged_with(tag)
+  }
+  scope :by_kind, lambda { |kind|
+    return all unless kind.present?
+    where(kind: kind)
+  }
+  scope :by_collection, lambda { |collection|
+    return all unless collection.present?
+    where(collection: collection)
+  }
+  scope :with_order, lambda { |by|
```

Comment:
We may as well handle order as a named scope, just like we do with the where clauses.